src/main/java/ai/promoted/java/client/TwoArmExperiment.java

```diff
@@ -0,0 +1,142 @@
+  
+  private final int numTotalBuckets;
+
+  /**
+   * Factory method for a 50/50 experiment.
+   * @return the configured experiment
+   */
+  public static TwoArmExperiment create5050TwoArmExperimentConfig(String cohortId, int controlPercent,
+      int treatmentPercent) {
+    if (controlPercent < 0 || controlPercent > 100) {
+      throw new IllegalArgumentException("Control percent must be in the range [0, 100]");
+    }
+    if (treatmentPercent < 0 || treatmentPercent > 100) {
+      throw new IllegalArgumentException("Treatment percent must be in the range [0, 100]");
+    }
+    return new TwoArmExperiment(cohortId, controlPercent, 100, treatmentPercent, 100);
+  }
+  
+  /**
+   * Creates a two-arm experiment config with the given parameters.
+   * @param cohortId the cohort (experiment) name
+   * @param numActiveControlBuckets number of active control buckets
+   * @param numControlBuckets total number of control buckets
+   * @param numActiveTreatmentBuckets number of active treatment buckets
+   * @param numTreatmentBuckets total number of treatment buckets
+   */
+  public TwoArmExperiment(String cohortId, int numActiveControlBuckets, int numControlBuckets,
+      int numActiveTreatmentBuckets, int numTreatmentBuckets) {
+    if (cohortId == null || cohortId.isBlank()) {
+      throw new IllegalArgumentException("Cohort ID must be non-empty");
+    }
+    
+    if (numControlBuckets < 0) {
+      throw new IllegalArgumentException("Control buckets must be positive");
+    }
+    
+    if (numTreatmentBuckets < 0) {
+      throw new IllegalArgumentException("Treatment buckets must be positive");
+    }
+    
+    if (numActiveControlBuckets < 0 || numActiveControlBuckets > numControlBuckets) {
+      throw new IllegalArgumentException("Active control buckets must be between 0 and the total number of control buckets");
+    }
+    
+    if (numActiveTreatmentBuckets < 0 || numActiveTreatmentBuckets > numTreatmentBuckets) {
+      throw new IllegalArgumentException("Active treatment buckets must be between 0 and the total number of treatment buckets");
+    }
+    
+    this.cohortId = cohortId;
+    this.cohortIdHash = cohortId.hashCode();
+    this.numActiveControlBuckets = numActiveControlBuckets;
+    this.numControlBuckets = numControlBuckets;
+    this.numActiveTreatmentBuckets = numActiveTreatmentBuckets;
+    this.numTreatmentBuckets = numTreatmentBuckets;
+    this.numTotalBuckets = numTreatmentBuckets + numControlBuckets;
+  }
+
+  public String getCohortId() {
+    return cohortId;
+  }
+
+  public int getNumActiveControlBuckets() {
+    return numActiveControlBuckets;
+  }
+
+  public int getNumControlBuckets() {
+    return numControlBuckets;
+  }
+
+  public int getNumActiveTreatmentBuckets() {
+    return numActiveTreatmentBuckets;
+  }
+
+  public int getNumTreatmentBuckets() {
+    return numTreatmentBuckets;
+  }
+  
+  public CohortMembership checkMembership(String userId) {
```

Comment:
Comment that this can return null if the user isn't activated into an experiment.